src/compiler/checker.ts

```diff
@@ -17736,7 +17737,19 @@ namespace ts {
                     // this regardless of the number of signatures, but the potential costs are prohibitive due
                     // to the quadratic nature of the logic below.
                     const eraseGenerics = relation === comparableRelation || !!compilerOptions.noStrictGenericChecks;
-                    result = signatureRelatedTo(sourceSignatures[0], targetSignatures[0], eraseGenerics, reportErrors, incompatibleReporter(sourceSignatures[0], targetSignatures[0]));
+                    const sourceSignature = first(sourceSignatures);
+                    const targetSignature = first(targetSignatures);
+                    result = signatureRelatedTo(sourceSignature, targetSignature, eraseGenerics, reportErrors, incompatibleReporter(sourceSignature, targetSignature));
+                    if (!result && reportErrors && kind === SignatureKind.Construct && (sourceObjectFlags & targetObjectFlags)) {
+                        const declaration = targetSignature.declaration || sourceSignature.declaration;
+                        if (declaration && (isConstructorDeclaration(declaration) || isConstructSignatureDeclaration(declaration))) {

```

Comment:
I don't get why `isConstructSignatureDeclaration` is desirable here. Will you get the same issues with the following?

```ts
declare let x: { new() => string };
declare let y: { new() => boolean };
```

Reply:
I think using `isConstructorDeclaration` would be enough for this case.